Source/ARTPush.m

```diff
@@ -98,14 +100,8 @@ - (dispatch_queue_t)queue {
 
 #if TARGET_OS_IOS
 
-// Store address of once_token to access it in debug function.
-static dispatch_once_t *activationMachine_once_token;
-
 - (ARTPushActivationStateMachine *)activationMachine {
-    static dispatch_once_t once;
-    activationMachine_once_token = &once;
-    static id activationMachineInstance;
-    dispatch_once(&once, ^{
+    if (_activationMachine == nil) {
```

Comment:
Should we concerned about the potential that this method might be called from multiple threads?

Reply:
This is an internal method which is only called from the library's GCD queue.

Generally, the library is architected as a public facade with an "internal" counterpart. The public facade wraps calls to the internal counterpart in GCD dispatches to a single, serial queue.